tinygrad/device.py

```diff
@@ -278,6 +279,9 @@ def get_linearizer(self, *ast:LazyOp) -> Linearizer:
         timed = sorted([(nm, tk, time_linearizer(tk, test_rawbuffers, allow_test_size=False, clear_l2=True)) for nm, tk in lins], key=lambda x: x[2])
         if DEBUG >= 1: print("  <  ".join(f"{nm:6s} : {lin.colored_shape(30, dense=True)} : {tm*1e6:8.2f} us" for nm, lin, tm in timed))
         k = timed[0][1]
+        if logkern is not None and logkern_level > 1: logkern.writelines(map(lambda x: str((x[1].ast, x[1].applied_opts,))+"\n", timed[1:]))
```

Comment:
nit: i think `logkern.writelines([f"{(lin.ast, lin.applied_opts)}\n" for _,lin in timed])` is cleaner. similar below

also are you sure you want to drop timed[0] here? that's not necessarily the best kernel

Reply:
will fix both to use f string.  when in BEAM, timed[0] will be written in line 284 below because it's assigned to k at line 281.

LOGKERN_LEVEL=1 will write just the kernel that is eventually returned, and LOGKERN_LEVEL=2 will write all timed ones.

Reply:
oh this is other 1 or 2.... yea you are right. somehow i thought this is other beam candidates